data/lescure2020clinical/lescure2020clinical-extraction.md

```diff
@@ -0,0 +1,85 @@
+# Extraction of the raw data for Lescure et al. (2020) using the data published by Goyal et al. (2020)
+
+[Goyal et al. (2020)](https://www.science.org/doi/10.1126/sciadv.abc7112) developed mathematical models to project multiple therapeutic approaches. The authors used four datasets of SARS-CoV-2 shedding in the absence of effective treatment to develop and validate a mathematical model. These data included including a dataset of 4 patients from France ([Lescure et al. 2020](https://www.thelancet.com/journals/laninf/article/PIIS1473-3099(20)30200-0/fulltext)).
+
+First, we `import` python modules needed:
+
+```python
+#import modules;
+import yaml
+import pandas as pd
+
+#functions to add folded blocks and literal blocks;
+class folded_str(str): pass
+class literal_str(str): pass
+
+def folded_str_representer(dumper, data):
+    return dumper.represent_scalar(u'tag:yaml.org,2002:str', data, style='>')
+def literal_str_representer(dumper, data):
+    return dumper.represent_scalar(u'tag:yaml.org,2002:str', data, style='|')
+
+yaml.add_representer(folded_str, folded_str_representer)
+yaml.add_representer(literal_str, literal_str_representer)
+```
+
+Raw data ([Viral_Loads](https://github.com/shedding-hub/shedding-hub/blob/main/data/lescure2020clinical/Viral_Loads.csv)), which is stored on [Shedding Hub](https://github.com/shedding-hub), will be loaded and cleaned to match the most updated [schema](https://github.com/shedding-hub/shedding-hub/blob/main/data/.schema.yaml).
+
+```python
+#load the data;
+Lescure2020 = pd.read_csv("Viral_Loads.csv")
+#Subset the data for Lescure et al. 2020;
+Lescure2020 = Lescure2020[Lescure2020["cov_study"]==4]
+#correct some doa errors;
+Lescure2020.loc[(Lescure2020["ID"]=="E3") & (Lescure2020["dao"]==13),"dao"]=14
+Lescure2020.loc[(Lescure2020["ID"]=="E3") & (Lescure2020["dao"]==15),"dao"]=16
+#some data cleaning to match the schema;
+Lescure2020.loc[Lescure2020["cens"]==0,"VL"]=10**Lescure2020.loc[Lescure2020["cens"]==0,"VL"]
+Lescure2020.loc[Lescure2020["cens"]==1,"VL"]="negative"
+#The positive but not quantifiable data are from Figure 2 in the paper;
+Lescure2020.loc[(Lescure2020["ID"]=="E3") & (Lescure2020["dao"].isin([14,16])),"VL"]="positive"
+Lescure2020.loc[(Lescure2020["ID"]=="E5") & (Lescure2020["dao"]==8),"VL"]="positive"
+
+#add data from the patient 2;
```

Comment:
Why is this patient separate from all the other ones in `Viral_Loads.csv`?

Reply:
The `Viral_Loads.csv` is the data from Goyal et al. (2020), which only included 4 patients from Lescure et al. (2020). Data from Patient 2 were added based on Figure 2 in Lescure et al. (2020). I added this in the comment.